pkgs/top-level/all-packages.nix

```diff
@@ -5003,7 +5005,8 @@ with pkgs;
   texFunctions = callPackage ../tools/typesetting/tex/nix pkgs;
 
   # TeX Live; see https://nixos.org/nixpkgs/manual/#sec-language-texlive
-  texlive = recurseIntoAttrs (callPackage ../tools/typesetting/tex/texlive { });
+  texlive = recurseIntoAttrs (callPackage ../tools/typesetting/tex/texlive/frozen.nix { });
```

Comment:
Picking up on the above, I believe tradition is to use `default.nix` for the stable version (thus `default.nix -> make-tex-live.nix` and `frozen.nix -> default.nix`).

Reply:
There already are a lot of files here and I feel like this makes it even less obvious where what is as this breaks up the symmetry between the two versions and replaces `default.nix` with a mostly empty file just containing version information. How do you feel about the following layout:

```
├── bin.nix
├── combine.nix
├── default.nix
├── generate-fixed-hashes.nix
├── latest
│   ├── fixed-hashes.nix
│   ├── tlpdb.nix
│   └── version-info.nix
├── make-texlive-packages.nix
├── stable
│   ├── fixed-hashes.nix
│   ├── tlpdb.nix
│   └── version-info.nix
├── tl2nix.sed
└── UPGRADING.md
```

Clearly version specific stuff is put into subdirectories whereas `default.nix` becomes a mostly generic entry point that builds the texlive(_latest) scope. By default it imports the version-info from the stable subdirectory, but this is overridden in `all-packages.nix` for the latest version. The code that builds the texlive packages (currently `tl` in `default.nix` is extracted into a separate file `make-texlive-packages`.

Reply:
@apfelkuchen6 now that the binary containers are in, what do you think of refactoring the packages in `texlivePackages`? Presumably with `tlpdb.nix` and `fixed-hashes.nix` in a separate `texlive-packages/` folder, while `texlive` keeps the combine machinery in place. The Nix expressions to build the packages should probably stay under `texlive`, e.g. `texlive.buildPackage`. Priority should be to make the entire thing easy to override, something which is currently very hard.

You have a few pieces here already – shall we start from those? Say with a new 0-rebuild PR?